spec/spec_models.cr

```diff
@@ -0,0 +1,154 @@
+class Granite::ORM::Base
+  def self.drop_and_create
+  end
+end
+
+{% for adapter in GraniteExample::ADAPTERS %}
+  {%
+    adapter_const_suffix = adapter.camelcase.id
+    adapter_suffix = "_#{adapter.id}".id
+    adapter_literal = adapter.id
+
+    parent_table = "parent_#{ adapter_literal }s".id
+    student_table = "student_#{ adapter_literal }s".id
+    teacher_table = "teacher_#{ adapter_literal }s".id
+    klass_table = "klass_#{ adapter_literal }s".id
+    enrollment_table = "enrollment_#{ adapter_literal }s".id
+
+    parent_class = "Parent#{ adapter_const_suffix }".id
+    student_class = "Student#{ adapter_const_suffix }".id
+    teacher_class = "Teacher#{ adapter_const_suffix }".id
+
+    if adapter == "pg"
+      primary_key_sql = "BIGSERIAL PRIMARY KEY".id
+      foreign_key_sql = "BIGINT".id
+    elsif adapter == "mysql"
+      primary_key_sql = "BIGINT NOT NULL AUTO_INCREMENT PRIMARY KEY".id
+      foreign_key_sql = "BIGINT".id
+    elsif adapter == "sqlite"
+      primary_key_sql = "INTEGER NOT NULL PRIMARY KEY".id
+      foreign_key_sql = "INTEGER".id
+    end
```

Comment:
why don't you use `case` instead ?

```
case adapter
when "pg"
when "mysql"
when "sqlite"
end
```

Reply:
Good question! Unless something has changed, case statements don’t work in macros.